From da73eca6714d670dd5c77803b5432239dd092bb8 Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Thu, 30 Aug 2018 17:07:53 -0500
Subject: [PATCH 4507/5242] MLK-18859-3: mtd: fsl-quadspi: resize the AHB
 buffer size

commit  5cd5f60095a667ec511865cb1fa6126a7be4de0f from
https://source.codeaurora.org/external/imx/linux-imx.git

decrese the AHB buffer size for AHB read as a workaround for random
UBIFS data corruption

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/spi-nor/fsl-quadspi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index 4a1433e..27245c3 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -742,7 +742,7 @@ static int fsl_qspi_init_ahb_read(struct fsl_qspi *q)
 	 * read performance.
 	 */
 	qspi_writel(q, QUADSPI_BUF3CR_ALLMST_MASK |
-			((q->devtype_data->ahb_buf_size / 8)
+			((q->devtype_data->ahb_buf_size / 8 / 8)
 			<< QUADSPI_BUF3CR_ADATSZ_SHIFT),
 			base + QUADSPI_BUF3CR);
 
-- 
1.7.9.5

